core/cas-server-core-authentication-mfa/src/main/java/org/apereo/cas/authentication/DefaultMultifactorAuthenticationContextValidator.java

```diff
@@ -60,11 +62,14 @@
             LOGGER.debug("No multifactor authentication providers are configured");
             return Pair.of(Boolean.FALSE, Optional.empty());
         }
-        final Optional<MultifactorAuthenticationProvider> requestedProvider = locateRequestedProvider(providerMap.values(), requestedContext);
+        final Collection<MultifactorAuthenticationProvider> flattenedProviders = MultifactorAuthenticationUtils.flattenProviders(providerMap.values());
+        LOGGER.debug("Available MFA Providers are [{}]", flattenedProviders);
+        final Optional<MultifactorAuthenticationProvider> requestedProvider = locateRequestedProvider(flattenedProviders, requestedContext);
         if (!requestedProvider.isPresent()) {
             LOGGER.debug("Requested authentication provider cannot be recognized.");
             return Pair.of(Boolean.FALSE, Optional.empty());
         }
+        LOGGER.debug("RequestedContext is [{}] and Available Contexts are [{}]", requestedContext, contexts);
```

Comment:
`LOGGER.debug("Requested authentication context is [{}] and available contexts are [{}]", requestedContext, contexts);`